### PR TITLE
replica: ignore cleanup of deallocated storage group

### PR DESCRIPTION
**Pull request by Deexie**
Cleanup of a deallocated tablet throws an exception. 
Since failed cleanup is retried, we end up in an infinite loop.

Ignore cleanup of deallocated storage groups.

Fixes:  #19752.

Needs to be backported to all branches with tablets (6.0 and later)

### CODE REVIEW

**Review by denesb — APPROVED** [review]

File: replica/table.cc [hunk 1]
```diff
@@ -3688,13 +3693,16 @@ future<> table::cleanup_compaction_groups(database& db, db::system_keyspace& sys
...
-    _sg_manager->remove_storage_group(tid.value());
+    if (auto* sg = _sg_manager->maybe_storage_group_for_id(_schema, tid.value())) {
+        co_await clear_inactive_reads_for_tablet(db, *sg);
+        // compaction_group::stop takes care of flushing.
+        co_await stop_compaction_groups(*sg);
+        co_await cleanup_compaction_groups(db, sys_ks, tid, *sg);
+        _sg_manager->remove_storage_group(tid.value());
+        co_return;
+    }
+    tlogger.warn("Storage group for tablet {} is deallocated. Ignore cleanup.", tid);
```
Comment (bhalevy) [inline replica/table.cc:3705]: To reduce the code churn, you could reverse the logic order, on these line:
```
auto sgp = _sg_manager->maybe_storage_group_for_id(_schema, tid.value());
if (!sgp) {
    tlogger.warn("Storage group for tablet {} is deallocated. Ignore cleanup.", tid);
    co_return;
}

auto& sg = *sgp;
...
```

Comment (bhalevy) [reply replica/table.cc:3705]: It's a bit odd to put the error-free path in a block that co_return's and then have a warning in the "default" path (which is actually the benign error path)

**Comment by Deexie** [comment]
- reverse the cleanup_tablet logic

**Review by bhalevy — APPROVED** [review]

**Review by raphaelsc — APPROVED** [review]

**Comment by bhalevy** [comment]
@scylladb/scylla-maint please consider merging

**Pull request merged**